src/components/user/user.service.ts

```diff
@@ -401,15 +400,27 @@ export class UserService {
 
     // Update roles
     if (input.roles) {
-      const newRoles = difference(input.roles, user.roles.value);
       await this.db
         .query()
         .match([
           node('user', ['User', 'BaseNode'], {
             id: input.id,
           }),
+          relation('out', 'oldRoleRel', 'roles'),
+          node('oldRoles', 'Property'),
         ])
-        .create([...this.roleProperties(newRoles)])
+        .delete('oldRoleRel')
+        .delete('oldRoles')
+        .run();
```

Comment:
@michaelmarshall you still want active=false here right?

Reply:
yes `active` still exists on relationships

Reply:
so don't `delete` relationships, just mark them as `active = false`